exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -47,6 +49,14 @@
   public List<String> files;
   public final String selectionRoot;
 
+  private enum StatusType {
+    CHECKED_DIRS,        // whether we have already checked for directories
+    HAS_DIRS,            // whether directories were found in the selection
+    EXPANDED_DIRS        // whether this selection has been expanded to files
+  }
+
+  private final BitSet dirStatus;
```

Comment:
Yeah, there was probably some disconnect since your previous comment  was a little succinct :).  I was trying to keep track of multiple states, but agree it can be simplified.  I may not need to use enumerator constructors. I will post an updated patch once I get a clean test run.